lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs

```diff
@@ -560,31 +573,88 @@ toCardanoTxOut (W.TxOut addr coin) =
 -------------------------------------------------------------------------------}
 
 instance EncodeAddress 'Mainnet where
-    encodeAddress = T.decodeUtf8 . hex . W.unAddress
+    encodeAddress = _encodeAddress
 
 instance EncodeAddress ('Testnet pm) where
-    encodeAddress = T.decodeUtf8 . hex . W.unAddress
+    encodeAddress = _encodeAddress
 
-_decodeAddress :: Text -> Either TextDecodingError W.Address
-_decodeAddress x = validateWithLedger =<< W.Address <$> fromHex x
+_encodeAddress :: W.Address -> Text
+_encodeAddress (W.Address bytes) =
+    if isJust (CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes)
```

Comment:
A golden here would be nice; I don't think a mistake here would be caught

Reply:
I agree. I intended to add some tests, hence the _draft_ :+1: